confgenerator/testdata/valid/windows/metrics-no_conf/input.yaml

```diff
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 metrics:
-  service:
-    pipelines:
-      default_pipeline:
-        receivers: []
-        exporters: []
+  processors:
```

Comment:
For disabling the entire metrics side, does it make sense to do this instead? It saves some resource (scraping, filtering and dropping metrics data points take CPU cycles.

```
metrics:
  service:
    pipelines:
      default_pipeline: []
```

Reply:
Can we not optimize this later (i.e., if we have a filter that drops everything, don't even bother generating the receivers on the otel side)?

Reply:
I don't believe that will work (since `default_pipeline` isn't a list), but also, this is optimization. We can always do that later without changing the syntax.

Reply:
Does

```
metrics:
  service:
    pipelines:
      default_pipeline:
        receivers: []
        exporters: []
```

still work if the agent for some reason has an IIS bug on the scraper side that caused it to crash? (We can avoid advertising too much for it, but leaving the option on the table for support case mitigation)?

Reply:
Confirmed that works:

```
lingshi@lingshi:~/repo/ops-agent/confgenerator/testdata/valid/linux/metrics-no_conf$ cat input.yaml
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
metrics:
  service:
    pipelines:
      default_pipeline:
        receivers: []
        exporters: []
lingshi@lingshi:~/repo/ops-agent/confgenerator/testdata/valid/linux/metrics-no_conf$ cat merged-config.yaml
logging:
  receivers:
    syslog:
      type: files
      include_paths:
      - /var/log/messages
      - /var/log/syslog
  exporters:
    google:
      type: google_cloud_logging
  service:
    pipelines:
      default_pipeline:
        receivers:
        - syslog
        exporters:
        - google
metrics:
  receivers:
    hostmetrics:
      type: hostmetrics
      collection_interval: 60s
  processors:
    metrics_filter:
      type: exclude_metrics
      metrics_pattern: []
  exporters:
    google:
      type: google_cloud_monitoring
  service:
    pipelines:
      default_pipeline:
        receivers: []
        processors: []
        exporters: []
```